source/components/select/select-component.js

```diff
@@ -117,9 +124,17 @@ class Select extends Component {
     return currentOption.length ? currentOptionValues : false;
   }
 
+  clearFilterInput() {
+    if (this.inputFilter) {
+      this.setState({ filteredItems: this.props.items });
+      this.inputFilter.value = null;
+    }
+  }
```

Comment:
Função para ser acionada toda vez que o filtro precisar ser limpo (quando houver clique fora do Select ou quando alguma opção for selecionada)